### ISSUE TITLE
garden-util-daemon pods not respecting nodeSelector

### ISSUE THREAD

**Issue by ghost** [garden-io/garden#2187]
### Current Behavior

garden-util-daemon pods are not respecting nodeSelector.

### Expected behavior

garden-util-daemon respects configured nodeSelector.

### Reproducible example

~~~
environments:
  - name: dev
    providers:
      - name: kubernetes
        context: ops
        buildMode: kaniko
        deploymentRegistry:
          hostname: index.docker.io
          namespace: example
        imagePullSecrets:
          - name: docker-hub
            namespace: garden-secrets
        systemNodeSelector:
          workload: apps
~~~

the generated chart values for util doesn't have the nodeSelector injected compared to say build-sync.

### Workaround

None at the moment.

### Your environment

* **OS:** macOS
* **How I'm running Kubernetes:** AKS

garden 0.12.12

**Comment by eysi09** [comment]
There's actually an open PR for this. But I suppose we can let stale bot do its thing.

**Comment by vvagaytsev** [comment]
Fixed in #3391
